### PR TITLE
Use new eth binary for trusty.

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/3649 (hopefully)

### CODE REVIEW

**Comment by chriseth** [comment]
Phew, that seems to have worked!

File: scripts/tests.sh [hunk 1]
```diff
@@ -63,8 +63,8 @@ function download_eth()
         ETH_HASH="e527dd3e3dc17b983529dd7dcfb74a0d3a5aed4e"
         if grep -i trusty /etc/lsb-release >/dev/null 2>&1
         then
-            ETH_BINARY=eth_byzantium2
-            ETH_HASH="4dc3f208475f622be7c8e53bee720e14cd254c6f"
+            ETH_BINARY=eth_byzantium_trusty
```
Comment (axic) [inline scripts/tests.sh]: Can we put here a cpp-ethereum commit hash this was built from?

Comment (chriseth) [reply scripts/tests.sh]: Done (we can squash while merging)

Comment (axic) [reply scripts/tests.sh]: Also is the artful build above the same or is it different?

**Review by axic — APPROVED** [review]

File: scripts/tests.sh (hunk 1 above)
Comment (chriseth) [reply scripts/tests.sh]: It is a different build, I do not remember which one exactly. I made is together with enabling endtoend tests on circle.

**Pull request merged**